dev/system-test/pipeline.ts

```diff
@@ -0,0 +1,733 @@
+          lt(Field.of('rating'), 4.5),
+          neq('genre', 'Science Fiction')
+        )
+      )
+      .select('rating', 'title')
+      .sort(Field.of('title').ascending())
+      .execute();
+    expectResults(
+      results,
+      {rating: 4.3, title: 'Crime and Punishment'},
+      {
+        rating: 4.3,
+        title: 'One Hundred Years of Solitude',
+      },
+      {rating: 4.5, title: 'Pride and Prejudice'}
+    );
+  });
+
+  it('testLogicalOperators', async () => {
+    const results = await randomCol
+      .pipeline()
+      .where(
+        or(
+          and(gt('rating', 4.5), eq('genre', 'Science Fiction')),
+          lt('published', 1900)
+        )
+      )
+      .select('title')
+      .sort(Field.of('title').ascending())
+      .execute();
+    expectResults(
+      results,
+      {title: 'Crime and Punishment'},
+      {title: 'Dune'},
+      {title: 'Pride and Prejudice'}
+    );
+  });
+
+  it('testChecks', async () => {
+    const results = await randomCol
+      .pipeline()
+      .where(not(Field.of('rating').isNaN()))
+      .select(
+        isNull('rating').as('ratingIsNull'),
+        not(Field.of('rating').isNaN()).as('ratingIsNotNaN')
+      )
+      .limit(1)
+      .execute();
+    expectResults(results, {ratingIsNull: false, ratingIsNotNaN: true});
+  });
+
+  it('testMapGet', async () => {
+    const results = await randomCol
+      .pipeline()
+      .select(
+        Field.of('awards').mapGet('hugo').as('hugoAward'),
+        Field.of('title')
+      )
+      .where(eq('hugoAward', true))
+      .execute();
+    expectResults(
+      results,
+      {hugoAward: true, title: "The Hitchhiker's Guide to the Galaxy"},
+      {hugoAward: true, title: 'Dune'}
+    );
+  });
+
+  // it('testParent', async () => {
+  //   const results = await randomCol
+  //       .pipeline()
+  //       .select(
+  //           parent(randomCol.doc('chile').collection('subCollection').path).as(
+  //               'parent'
+  //           )
+  //       )
+  //       .limit(1)
+  //       .execute();
+  //   expect(results[0].data().parent.endsWith('/books')).to.be.true;
+  // });
+  //
+  // it('testCollectionId', async () => {
+  //   const results = await randomCol
+  //       .pipeline()
+  //       .select(collectionId(randomCol.doc('chile')).as('collectionId'))
+  //       .limit(1)
+  //       .execute();
+  //   expectResults(results, {collectionId: 'books'});
+  // });
+
+  it('testDistanceFunctions', async () => {
+    const sourceVector = [0.1, 0.1];
+    const targetVector = [0.5, 0.8];
+    const results = await randomCol
+      .pipeline()
+      .select(
+        cosineDistance(Constant.ofVector(sourceVector), targetVector).as(
+          'cosineDistance'
+        ),
+        dotProductDistance(Constant.ofVector(sourceVector), targetVector).as(
+          'dotProductDistance'
+        ),
+        euclideanDistance(Constant.ofVector(sourceVector), targetVector).as(
+          'euclideanDistance'
+        )
```

Comment:
This pattern will support KNN. If eventually there is ANN support we may need a different pattern. Do we need to consider that now?